### PR TITLE
[FLINK-36510][rpc] bump pekko to 1.1.2, remove netty 3 (backport to release-1.20)

### PR DESCRIPTION
**Pull request by gracegrimwood**
## What is the purpose of the change

Updates Pekko dependency to 1.1.2 which in turn upgrades Netty 3 to 4 (addressing [FLINK-29065](https://issues.apache.org/jira/browse/FLINK-29065) and removing several CVEs from Flink). Pekko 1.1 also upgrades other dependencies such as slf4j and Jackson. For more details see the [Pekko 1.1 release notes](https://pekko.apache.org/docs/pekko/current/release-notes/releases-1.1.html). CC @ferenc-csaky :smile:

## Brief change log

  - Update Pekko version from 1.0.1 to 1.1.2
  - Replace Pekko Netty 3 dependency with flink-shaded Netty 4 (and test-scoped direct Netty 4 dependency)
  - Update Netty imports in `ActorSystemBootstrapTools` and `PekkoUtils`
  - Update NOTICE file

## Verifying this change

This change is covered by existing tests in flink-rpc.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): yes
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): don't know
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no
## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-rpc/flink-rpc-akka/src/main/resources/META-INF/NOTICE [hunk 1]
```diff
@@ -6,17 +6,17 @@ The Apache Software Foundation (http://www.apache.org/).
...
 - com.typesafe:ssl-config-core_2.12:0.6.1
-- io.netty:netty:3.10.6.Final
-- org.agrona:agrona:1.15.1
-- org.apache.pekko:pekko-actor_2.12:1.0.1
-- org.apache.pekko:pekko-remote_2.12:1.0.1
-- org.apache.pekko:pekko-pki_2.12:1.0.1
-- org.apache.pekko:pekko-protobuf-v3_2.12:1.0.1
-- org.apache.pekko:pekko-slf4j_2.12:1.0.1
-- org.apache.pekko:pekko-stream_2.12:1.0.1
+- io.netty:netty-all:4.1.100.Final
```
Comment (ferenc-csaky) [inline flink-rpc/flink-rpc-akka/src/main/resources/META-INF/NOTICE:12]: This should be changed to `4.1.91.Final` according to the [root POM](https://github.com/apache/flink/blob/60cba350d7638592ea771dc7cf512798e6248886/pom.xml#L369C14-L369C26).

**Comment by ferenc-csaky** [comment]
1.20 backport of https://github.com/apache/flink/pull/25494, will merge them together.

File: flink-rpc/flink-rpc-akka/src/main/resources/META-INF/NOTICE (hunk 1 above)
Comment (gracegrimwood) [reply flink-rpc/flink-rpc-akka/src/main/resources/META-INF/NOTICE]: Nice catch! Changed this now :)

**Review by ferenc-csaky — APPROVED** [review]

**Comment by He-Pin** [comment]
@ferenc-csaky I'm so sorry, but I think I have fond the root cause. a fix is pending in https://github.com/apache/pekko/pull/1635

**Pull request merged**